rflx/pyrflx/typevalue.py

```diff
@@ -0,0 +1,196 @@
+from abc import abstractmethod, abstractproperty
+from typing import Any, Union
+
+from rflx.expression import TRUE, Expr, Variable
+from rflx.model import Enumeration, ModularInteger, Number, Opaque, RangeInteger, Scalar, Type
+
+
+class NotInitializedError(Exception):
+    pass
+
+
+class TypeValue:
+
+    _value: Any
+
+    def __init__(self, vtype: Type) -> None:
+        self._type = vtype
+        self._initialized = False
+
+    def __repr__(self) -> str:
+        args = ", ".join([f"{k}={v}" for k, v in self.__dict__.items()])
+        return f"{self.__class__.__name__}({args})"
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, self.__class__):
+            return (
+                self._initialized == other._initialized
+                and (
+                    self._value == other._value
+                    if (self._initialized and other._initialized)
+                    else True
+                )
+                and self._type == other._type
+            )
+        return False
+
+    @property
+    def initialized(self) -> bool:
+        return self._initialized
+
+    def _raise_initialized(self) -> None:
+        if not self._initialized:
+            raise NotInitializedError("value not initialized")
+
+    @abstractmethod
+    def assign(self, value: Any, check: bool) -> None:
+        raise NotImplementedError
+
+    @abstractproperty
+    def value(self) -> Any:
+        return NotImplemented
+
+    @abstractproperty
+    def binary(self) -> str:
+        return NotImplemented
+
+    @abstractproperty
+    def accepted_type(self) -> type:
+        return NotImplemented
+
+    @classmethod
+    def construct(cls, vtype: Type) -> "TypeValue":
+        if isinstance(vtype, (ModularInteger, RangeInteger)):
+            return IntegerValue(vtype)
+        if isinstance(vtype, Enumeration):
+            return EnumValue(vtype)
+        if isinstance(vtype, Opaque):
+            return OpaqueValue(vtype)
+        raise ValueError("cannot construct unknown type: " + repr(vtype))
```

Comment:
The error should rather contain the type name instead of the representation of `vtype`.

Reply:
Fixed by https://github.com/Componolit/RecordFlux/pull/158/commits/5477bfbe1b436d8cdd4fff9df15f9506c70e649f.